egs/wsj/s5/steps/make_phone_graph.sh

```diff
@@ -92,17 +92,14 @@ fi
 
 if [ $stage -le 3 ]; then
   echo "$0: creating G_phones.fst from ARPA"
-  gunzip -c $dir/phone_graph/arpa_noug.gz | arpa2fst - - | fstprint | \
-    utils/eps2disambig.pl | utils/s2eps.pl | \
-    awk '{if (NF < 5 || $5 < 100.0) { print; }}' | \
-    fstcompile --isymbols=$lang/phones.txt --osymbols=$lang/phones.txt \
-       --keep_isymbols=false --keep_osymbols=false | \
-    fstconnect | \
-    fstrmepsilon > $dir/phone_graph/G_phones.fst
-   fstisstochastic $dir/phone_graph/G_phones.fst  || echo "[info]: G_phones not stochastic."
+  gunzip -c $dir/phone_graph/arpa_noug.gz | \
+    arpa2fst --disambig-symbol=#0 --read-symbol-table=$lang/phones.txt - - | \
+    fstprint | awk '{if (NF < 5 || $5 < 100.0) { print; }}' | fstcompile | \
+    fstconnect > $dir/phone_graph/G_phones.fst
```

Comment:
OK thanks.

On Wed, Apr 13, 2016 at 2:48 PM, Kirill Katsnelson <notifications@github.com
> wrote:

> In egs/wsj/s5/steps/make_phone_graph.sh
> <https://github.com/kaldi-asr/kaldi/pull/682#discussion_r59632010>:
>
> > @@ -92,17 +92,14 @@ fi
> >
> >  if [ $stage -le 3 ]; then
> >    echo "$0: creating G_phones.fst from ARPA"
> > -  gunzip -c $dir/phone_graph/arpa_noug.gz | arpa2fst - - | fstprint | \
> > -    utils/eps2disambig.pl | utils/s2eps.pl | \
> > -    awk '{if (NF < 5 || $5 < 100.0) { print; }}' | \
> > -    fstcompile --isymbols=$lang/phones.txt --osymbols=$lang/phones.txt \
> > -       --keep_isymbols=false --keep_osymbols=false | \
> > -    fstconnect | \
> > -    fstrmepsilon > $dir/phone_graph/G_phones.fst
> > -   fstisstochastic $dir/phone_graph/G_phones.fst  || echo "[info]: G_phones not stochastic."
> > +  gunzip -c $dir/phone_graph/arpa_noug.gz | \
> > +    arpa2fst --disambig-symbol=#0 --read-symbol-table=$lang/phones.txt - - | \
> > +    fstprint | awk '{if (NF < 5 || $5 < 100.0) { print; }}' | fstcompile | \
> > +    fstconnect > $dir/phone_graph/G_phones.fst
>
> It was in every pipeline because of utils/s2eps.pl. The new invocation of
> arpa2fst with --disambig-symbol produces inherently epsilon-free machines.
> fstconnect does not add arcs, so fstrmepsilon is not needed.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/682/files/6e8dd7a3e849f08d04b4fb28cd3fbd772297d4bd#r59632010>
>